libraries/SPISlave/src/hspi_slave.c

```diff
@@ -85,7 +84,7 @@ void hspi_slave_begin(uint8_t status_len, void * arg)
     pinMode(MISO, SPECIAL);
     pinMode(MOSI, SPECIAL);
 
-    SPI1S = SPISE | SPISBE | 0x3E0; // SPI_SLAVE_REG
+    SPI1S = SPISE | SPISBE | SPISRBIE | SPISWBIE | SPISRSIE | SPISWSIE | SPISTRIE;	// 0x3E0 // SPI_SLAVE_REG
```

Comment:
These extras represent bits in this register (SPI1S). I have translated previous hardcoded value 0x3E0 to every bit setting separately (0x3E0 == SPISRBIE | SPISWBIE | SPISRSIE | SPISWSIE | SPISTRIE).
All these bits are defined here:
https://github.com/esp8266/Arduino/blob/master/cores/esp8266/esp8266_peri.h#L511-534

Reply:
I don't know why all of them used - I simply rewrite hard coded value for easier readability. But if it's really necessary I can try to research it.

Reply:
>  I have translated previous hardcoded value 0x3E0 to every bit setting separately

That's a very good thing you did!

> I can try to research it

What you did is already a step in the right direction. It would be enough to add a comment explaining where the meanings are.

Reply:
I have commented, I'm only not sure about TRANS interrupt - maybe it enables all other SPI interrupts, or it happens when any spi action detected. In [isr handler](https://github.com/esp8266/Arduino/blob/8ea1fb3b0651a7fae4b91184822e199a44238b58/libraries/SPISlave/src/hspi_slave.c#L31) this interrupt not used.